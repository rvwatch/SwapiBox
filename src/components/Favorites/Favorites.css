.favorites-container{
    margin: 120px auto auto auto;
    text-align: center;
  }

.no-favs-message{
    font-size: 70px;
    width: 300px;
    text-align: center;
    font-weight: lighter;
    color: #feda4a;
    margin: 100px auto auto auto;
    -webkit-text-fill-color: black;
    -webkit-text-stroke-width: 2px;
    -webkit-text-stroke-color: #feda4a;
}