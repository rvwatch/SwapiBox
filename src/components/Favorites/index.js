import React, { Component } from 'react';
import Card from '../Cards';

class Favorites extends Component {
  constructor() {
    super();

    this.state = {
      favorites: []
    };
  }

  buildCards = data => {
    if (this.props.favorites) {
      return data.map((fav, index) => (
        <Card
          type="favorites"
          setFavorites={this.props.setFavorites}
          selected={fav}
          {...fav}
          key={index}
        />
      ));
    }
  };

  render() {
    
    
    const createFavorites = this.buildCards(this.props.favorites);

    if (this.props.favorites.length) {
      return <section> {createFavorites} </section>;
    } else {
      return <section>
        <h1>There are no favorites!</h1>
      </section>;
    }
  }
}

export default Favorites;
