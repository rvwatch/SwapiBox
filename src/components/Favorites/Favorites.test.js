import React from 'react';
import { shallow } from 'enzyme';
import Favorites from './index';

describe('Favorites ', () => {
  let wrapper;
  let mockFavs;
  beforeEach(() => {
    mockFavs = [
      {
        data1: 'Terrain: ocean',
        data2: 'Population: 1000000000',
        data3: 'Climate: temperate',
        data4: 'Resident: Boba Fett,Lama Su,Taun We',
        favorite: true,
        title: 'Kamino'
      }
    ];

    wrapper = shallow(
      <Favorites favorites={mockFavs} setFavorites={mockFavs} />
    );
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly when given two cards', () => {
    mockFavs = [
      {
        data1: 'Terrain: ocean',
        data2: 'Population: 1000000000',
        data3: 'Climate: temperate',
        data4: 'Resident: Boba Fett,Lama Su,Taun We',
        favorite: true,
        title: 'Kamino'
      },
      {
        data1: 'Terrain: ocean',
        data2: 'Population: 1000000000',
        data3: 'Climate: temperate',
        data4: 'Resident: Boba Fett,Lama Su,Taun We',
        favorite: true,
        title: 'Kamino'
      }
    ];
    expect(wrapper).toMatchSnapshot();
  });
});
