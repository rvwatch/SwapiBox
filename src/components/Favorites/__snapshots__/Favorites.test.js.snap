// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites  renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Favorites
    favorites={
        Array [
            Object {
              "data1": "Terrain: ocean",
              "data2": "Population: 1000000000",
              "data3": "Climate: temperate",
              "data4": "Resident: Boba Fett,Lama Su,Taun We",
              "favorite": true,
              "title": "Kamino",
            },
          ]
    }
    setFavorites={
        Array [
            Object {
              "data1": "Terrain: ocean",
              "data2": "Population: 1000000000",
              "data3": "Climate: temperate",
              "data4": "Resident: Boba Fett,Lama Su,Taun We",
              "favorite": true,
              "title": "Kamino",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        " ",
        Array [
          <Card
            data1="Terrain: ocean"
            data2="Population: 1000000000"
            data3="Climate: temperate"
            data4="Resident: Boba Fett,Lama Su,Taun We"
            favorite={true}
            selected={
                        Object {
                                    "data1": "Terrain: ocean",
                                    "data2": "Population: 1000000000",
                                    "data3": "Climate: temperate",
                                    "data4": "Resident: Boba Fett,Lama Su,Taun We",
                                    "favorite": true,
                                    "title": "Kamino",
                                  }
            }
            setFavorites={
                        Array [
                                    Object {
                                      "data1": "Terrain: ocean",
                                      "data2": "Population: 1000000000",
                                      "data3": "Climate: temperate",
                                      "data4": "Resident: Boba Fett,Lama Su,Taun We",
                                      "favorite": true,
                                      "title": "Kamino",
                                    },
                                  ]
            }
            title="Kamino"
            type="favorites"
/>,
        ],
        " ",
      ],
      "className": "favorites-container",
    },
    "ref": null,
    "rendered": Array [
      " ",
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "data1": "Terrain: ocean",
          "data2": "Population: 1000000000",
          "data3": "Climate: temperate",
          "data4": "Resident: Boba Fett,Lama Su,Taun We",
          "favorite": true,
          "selected": Object {
            "data1": "Terrain: ocean",
            "data2": "Population: 1000000000",
            "data3": "Climate: temperate",
            "data4": "Resident: Boba Fett,Lama Su,Taun We",
            "favorite": true,
            "title": "Kamino",
          },
          "setFavorites": Array [
            Object {
              "data1": "Terrain: ocean",
              "data2": "Population: 1000000000",
              "data3": "Climate: temperate",
              "data4": "Resident: Boba Fett,Lama Su,Taun We",
              "favorite": true,
              "title": "Kamino",
            },
          ],
          "title": "Kamino",
          "type": "favorites",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      " ",
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          " ",
          Array [
            <Card
              data1="Terrain: ocean"
              data2="Population: 1000000000"
              data3="Climate: temperate"
              data4="Resident: Boba Fett,Lama Su,Taun We"
              favorite={true}
              selected={
                            Object {
                                          "data1": "Terrain: ocean",
                                          "data2": "Population: 1000000000",
                                          "data3": "Climate: temperate",
                                          "data4": "Resident: Boba Fett,Lama Su,Taun We",
                                          "favorite": true,
                                          "title": "Kamino",
                                        }
              }
              setFavorites={
                            Array [
                                          Object {
                                            "data1": "Terrain: ocean",
                                            "data2": "Population: 1000000000",
                                            "data3": "Climate: temperate",
                                            "data4": "Resident: Boba Fett,Lama Su,Taun We",
                                            "favorite": true,
                                            "title": "Kamino",
                                          },
                                        ]
              }
              title="Kamino"
              type="favorites"
/>,
          ],
          " ",
        ],
        "className": "favorites-container",
      },
      "ref": null,
      "rendered": Array [
        " ",
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "data1": "Terrain: ocean",
            "data2": "Population: 1000000000",
            "data3": "Climate: temperate",
            "data4": "Resident: Boba Fett,Lama Su,Taun We",
            "favorite": true,
            "selected": Object {
              "data1": "Terrain: ocean",
              "data2": "Population: 1000000000",
              "data3": "Climate: temperate",
              "data4": "Resident: Boba Fett,Lama Su,Taun We",
              "favorite": true,
              "title": "Kamino",
            },
            "setFavorites": Array [
              Object {
                "data1": "Terrain: ocean",
                "data2": "Population: 1000000000",
                "data3": "Climate: temperate",
                "data4": "Resident: Boba Fett,Lama Su,Taun We",
                "favorite": true,
                "title": "Kamino",
              },
            ],
            "title": "Kamino",
            "type": "favorites",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        " ",
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`Favorites  renders correctly when given two cards 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Favorites
    favorites={
        Array [
            Object {
              "data1": "Terrain: ocean",
              "data2": "Population: 1000000000",
              "data3": "Climate: temperate",
              "data4": "Resident: Boba Fett,Lama Su,Taun We",
              "favorite": true,
              "title": "Kamino",
            },
          ]
    }
    setFavorites={
        Array [
            Object {
              "data1": "Terrain: ocean",
              "data2": "Population: 1000000000",
              "data3": "Climate: temperate",
              "data4": "Resident: Boba Fett,Lama Su,Taun We",
              "favorite": true,
              "title": "Kamino",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        " ",
        Array [
          <Card
            data1="Terrain: ocean"
            data2="Population: 1000000000"
            data3="Climate: temperate"
            data4="Resident: Boba Fett,Lama Su,Taun We"
            favorite={true}
            selected={
                        Object {
                                    "data1": "Terrain: ocean",
                                    "data2": "Population: 1000000000",
                                    "data3": "Climate: temperate",
                                    "data4": "Resident: Boba Fett,Lama Su,Taun We",
                                    "favorite": true,
                                    "title": "Kamino",
                                  }
            }
            setFavorites={
                        Array [
                                    Object {
                                      "data1": "Terrain: ocean",
                                      "data2": "Population: 1000000000",
                                      "data3": "Climate: temperate",
                                      "data4": "Resident: Boba Fett,Lama Su,Taun We",
                                      "favorite": true,
                                      "title": "Kamino",
                                    },
                                  ]
            }
            title="Kamino"
            type="favorites"
/>,
        ],
        " ",
      ],
      "className": "favorites-container",
    },
    "ref": null,
    "rendered": Array [
      " ",
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "data1": "Terrain: ocean",
          "data2": "Population: 1000000000",
          "data3": "Climate: temperate",
          "data4": "Resident: Boba Fett,Lama Su,Taun We",
          "favorite": true,
          "selected": Object {
            "data1": "Terrain: ocean",
            "data2": "Population: 1000000000",
            "data3": "Climate: temperate",
            "data4": "Resident: Boba Fett,Lama Su,Taun We",
            "favorite": true,
            "title": "Kamino",
          },
          "setFavorites": Array [
            Object {
              "data1": "Terrain: ocean",
              "data2": "Population: 1000000000",
              "data3": "Climate: temperate",
              "data4": "Resident: Boba Fett,Lama Su,Taun We",
              "favorite": true,
              "title": "Kamino",
            },
          ],
          "title": "Kamino",
          "type": "favorites",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      " ",
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          " ",
          Array [
            <Card
              data1="Terrain: ocean"
              data2="Population: 1000000000"
              data3="Climate: temperate"
              data4="Resident: Boba Fett,Lama Su,Taun We"
              favorite={true}
              selected={
                            Object {
                                          "data1": "Terrain: ocean",
                                          "data2": "Population: 1000000000",
                                          "data3": "Climate: temperate",
                                          "data4": "Resident: Boba Fett,Lama Su,Taun We",
                                          "favorite": true,
                                          "title": "Kamino",
                                        }
              }
              setFavorites={
                            Array [
                                          Object {
                                            "data1": "Terrain: ocean",
                                            "data2": "Population: 1000000000",
                                            "data3": "Climate: temperate",
                                            "data4": "Resident: Boba Fett,Lama Su,Taun We",
                                            "favorite": true,
                                            "title": "Kamino",
                                          },
                                        ]
              }
              title="Kamino"
              type="favorites"
/>,
          ],
          " ",
        ],
        "className": "favorites-container",
      },
      "ref": null,
      "rendered": Array [
        " ",
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "data1": "Terrain: ocean",
            "data2": "Population: 1000000000",
            "data3": "Climate: temperate",
            "data4": "Resident: Boba Fett,Lama Su,Taun We",
            "favorite": true,
            "selected": Object {
              "data1": "Terrain: ocean",
              "data2": "Population: 1000000000",
              "data3": "Climate: temperate",
              "data4": "Resident: Boba Fett,Lama Su,Taun We",
              "favorite": true,
              "title": "Kamino",
            },
            "setFavorites": Array [
              Object {
                "data1": "Terrain: ocean",
                "data2": "Population: 1000000000",
                "data3": "Climate: temperate",
                "data4": "Resident: Boba Fett,Lama Su,Taun We",
                "favorite": true,
                "title": "Kamino",
              },
            ],
            "title": "Kamino",
            "type": "favorites",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        " ",
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
